### PR TITLE
If a WebExtension is submitted without an add-on id, create one.

### LINKED ISSUE #1891: If a WebExtension is submitted without an add-on id, create one
Allow a WebExtension to be submitted without an add-on id. If it is, then lets create one and insert it into the mozilla.json file and save it inside the add-on.

This might mean altering an already existing file mozilla.json if one is submitted by a developer, so we can't just overwrite the whole file.

Tracker: mozilla/addons#61

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #1891

I'm honestly not sure if I need more tests somewhere. I'm totally open to ideas.

### CODE REVIEW

File: src/olympia/files/tests/test_utils_.py [hunk 1]
```diff
@@ -325,8 +325,8 @@ def test_guid(self):
                     'id': 'some-id'}}})['guid'] == 'some-id'
 
     def test_name_for_guid_if_no_id(self):
-        """Use the name for the guid if there is no id."""
-        assert self.parse({'name': 'addon-name'})['guid'] == 'addon-name'
+        """Don't use the name for the guid if there is no id."""
+        assert self.parse({'name': 'addon-name'})['guid'] is None
```
Comment (EnTeQuAk) [inline src/olympia/files/tests/test_utils_.py]: According to the spec there's only `id` and `name` is the actual name of the add-on. So I changed it, with good intentions.

File: src/olympia/addons/tests/test_models.py [hunk 2]
```diff
@@ -2168,6 +2169,33 @@ def test_validation_timeout(self):
         addon = Addon.from_upload(upload, [self.platform])
         assert addon.admin_review
 
+    def test_webextension_guid_required(self):
+        """Test that the guid is required. This test should be removed once
+        the waffle-flag 'addons-linter' will be deleted.
```
Comment (tofumatt) [inline src/olympia/addons/tests/test_models.py]: Should we add a `TODO:` then?

**Comment by tofumatt** [comment]
r+

File: src/olympia/addons/models.py [hunk 3]
```diff
@@ -16,6 +17,7 @@
 from django.db.models import Max, Q, signals as dbsignals
 from django.utils.translation import trans_real, ugettext_lazy as _
 
+import waffle
```
Comment (tofumatt) [inline src/olympia/addons/models.py]: Nitpick, but should this be at the top? Seems we do imports in alphabetical order.

**Comment by EnTeQuAk** [comment]
Thanks for the review!

**Pull request merged**